tests/integration/PageListGetterTest.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Mediawiki\Api\Test;
+
+use Mediawiki\DataModel\Content;
+use Mediawiki\DataModel\PageIdentifier;
+use Mediawiki\DataModel\Revision;
+use Mediawiki\DataModel\Title;
+use PHPUnit_Framework_TestCase;
+
+/**
+ * Test the \Mediawiki\Api\Service\PageListGetter class.
+ */
+class PageListGetterTest extends PHPUnit_Framework_TestCase {
+
+	/** @var string */
+	private $emptyCatName = 'Category:Empty category';
+
+	/** @var string */
+	private $nonemptyCatName = 'Category:Test category';
+
+	/** @var \Mediawiki\Api\Service\PageListGetter */
+	private $pageListGetter;
+
+	/**
+	 * Set up some test categories and pages.
+	 */
+	public function setUp() {
+		$factory = TestEnvironment::newDefault()->getFactory();
+
+		// An empty category.
+		$emptyCat = new PageIdentifier(new Title($this->emptyCatName));
+		$factory->newRevisionSaver()->save(new Revision(new Content(''), $emptyCat));
+
+		// A non-empty category.
+		$testCat = new PageIdentifier(new Title($this->nonemptyCatName));
+		$factory->newRevisionSaver()->save(new Revision(new Content(''), $testCat));
+
+		// Some pages in the latter.
+		// (Count must exceed the default categorymember result set size of 10.)
+		for ($i = 1; $i <= 15; $i++) {
+			$testCat = new PageIdentifier(new Title("Test page $i"));
+			// Even pages link to Main Page, odd pages transclude {{test}}.
+			$mainPageLink = (($i % 2) == 0) ? 'Go to [[Main Page]].' : 'This is a {{test}}.';
+			$content = new Content("$mainPageLink\n\n[[$this->nonemptyCatName]]");
+			$factory->newRevisionSaver()->save(new Revision($content, $testCat));
+		}
+
+		$this->pageListGetter = $factory->newPageListGetter();
+	}
+
+	/**
+	 * @test
```

Comment:
Instead of adding @test annotations please just prefix the method with "test".
See https://github.com/addwiki/mediawiki-api/blob/master/tests/integration/UserIntegrationTest.php#L24 as an example.